plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/errormarker/FordiacMarkerHelper.java

```diff
@@ -101,20 +102,21 @@ public static String getTargetUriString(final IResource resource, final EObject
 		return null;
 	}
 
-	public static Object[] getDiagnosticData(final EObject object) {
-		return getDiagnosticData(object, null);
+	public static Object[] getDiagnosticData(final EObject object, final String... data) {
+		return getDiagnosticData(object, null, data);
 	}
 
-	public static Object[] getDiagnosticData(final EObject object, final EStructuralFeature feature) {
+	public static Object[] getDiagnosticData(final EObject object, final EStructuralFeature feature,
+			final String... data) {
 		if (object == null) {
-			return new Object[0];
+			return data;
 		}
-		return new Object[] { object, // [0] object itself (per EMF convention)
+		return Stream.concat(Stream.of(object, // [0] object itself (per EMF convention)
```

Comment:
can you add a comment with an example of such string ?

Reply:
I have added additional documentation on the resulting array.